### PR TITLE
Update POCL job to accept new postal order fields

### PR DESCRIPTION
**Pull request by irisfaraway**
https://eaflood.atlassian.net/browse/IWTF-3418

PR https://github.com/DEFRA/rod-licensing/pull/1713 added new fields to dynamics-lib to handle postal order payments producing validation errors. This PR updates the POCL job to include those new fields.

### CODE REVIEW

File: packages/pocl-job/src/staging/__tests__/pocl-validation-errors.spec.js [hunk 1]
```diff
@@ -59,107 +62,289 @@ describe('pocl-validation-errors', () => {
...
   })
 
   describe('processes successfully fixed validation errors', () => {
-    let poclValidationError
-    beforeEach(async () => {
-      poclValidationError = getPoclValidationError()
-      salesApi.getPoclValidationErrorsForProcessing.mockResolvedValue([poclValidationError])
-      salesApi.createTransactions.mockResolvedValue([{ statusCode: 201, response: { id: 'test-response-id' } }])
+    beforeAll(() => {
+      salesApi.getPoclValidationErrorsForProcessing.mockResolvedValue([getPoclValidationError()])
```
Comment (jaucourt) [inline packages/pocl-job/src/staging/__tests__/pocl-validation-errors.spec.js:87]: This needs to change to `mockImplementation` so a fresh value is available every time

File: packages/pocl-job/src/staging/__tests__/pocl-validation-errors.spec.js [hunk 2]
```diff
@@ -59,107 +62,325 @@ describe('pocl-validation-errors', () => {
...
       const validationError = salesApi.createTransactions.mock.calls[0][0][0]
-      expect(validationError).toMatchSnapshot()
+      const expectedDate = '2023-01-01T00:00:00Z'
+      expect(validationError.permissions[0].startDate).toBe(expectedDate)
     })
 
-    it('attempts to finalise transaction in the Sales Api', () => {
-      const record = salesApi.finaliseTransaction.mock.calls[0]
-      expect(record).toMatchSnapshot()
+    it('converts the startDate to ISO format without milliseconds', async () => {
```
Comment (irisfaraway) [inline packages/pocl-job/src/staging/__tests__/pocl-validation-errors.spec.js:360]: Can we tweak the name of this test so it is clear how it is different from the one on line 347?

File: packages/pocl-job/src/staging/__tests__/pocl-validation-errors.spec.js [hunk 3]
```diff
@@ -59,107 +62,325 @@ describe('pocl-validation-errors', () => {
...
+      const poclValidationError = getPoclValidationError({ serialNumber: null, newPaymentSource: { label: 'Postal Order' } })
+      salesApi.getPoclValidationErrorsForProcessing.mockResolvedValueOnce([poclValidationError])
+
+      await processPoclValidationErrors()
+
+      const [[[createTransactionPayload]]] = salesApi.createTransactions.mock.calls
+      expect(createTransactionPayload.serialNumber).toBe('Postal Order Sales')
+    })
+
+    it('leaves the datatype blank if payment source is not postal order', async () => {
```
Comment (irisfaraway) [inline packages/pocl-job/src/staging/__tests__/pocl-validation-errors.spec.js:278]:
```suggestion
    it('leaves the serial number blank if payment source is not postal order', async () => {
```

File: packages/pocl-job/src/staging/__tests__/pocl-validation-errors.spec.js [hunk 4]
```diff
@@ -59,107 +62,325 @@ describe('pocl-validation-errors', () => {
...
+      const poclValidationError = getPoclValidationError({ serialNumber: null, newPaymentSource: { label: 'Magic Beans' } })
+      salesApi.getPoclValidationErrorsForProcessing.mockResolvedValueOnce([poclValidationError])
+
+      await processPoclValidationErrors()
+
+      const [[[createTransactionPayload]]] = salesApi.createTransactions.mock.calls
+      expect(createTransactionPayload.serialNumber).toBeUndefined()
+    })
+
+    it('leaves the datatype blank if payment source is not set', async () => {
```
Comment (irisfaraway) [inline packages/pocl-job/src/staging/__tests__/pocl-validation-errors.spec.js:288]:
```suggestion
    it('leaves the serial number blank if payment source is not set', async () => {
```

File: packages/pocl-job/src/staging/__tests__/pocl-validation-errors.spec.js [hunk 5]
```diff
@@ -59,107 +62,325 @@ describe('pocl-validation-errors', () => {
...
+    it('fills it with the correct data for postal orders', async () => {
+      const poclValidationError = getPoclValidationError({ dataSource: null, newPaymentSource: { label: 'Postal Order' } })
+      salesApi.getPoclValidationErrorsForProcessing.mockResolvedValueOnce([poclValidationError])
+      await processPoclValidationErrors()
+
+      const [[[createTransactionPayload]]] = salesApi.createTransactions.mock.calls
+      expect(createTransactionPayload.dataSource).toBe('Postal Order Sales')
+    })
+
+    it('leaves the datatype blank if payment source is not postal order', async () => {
```
Comment (irisfaraway) [inline packages/pocl-job/src/staging/__tests__/pocl-validation-errors.spec.js:248]:
```suggestion
    it('leaves the dataSource blank if payment source is not postal order', async () => {
```

File: packages/pocl-job/src/staging/__tests__/pocl-validation-errors.spec.js [hunk 6]
```diff
@@ -59,107 +62,325 @@ describe('pocl-validation-errors', () => {
...
+      const poclValidationError = getPoclValidationError({ dataSource: null, newPaymentSource: { label: 'Magic beans' } })
+      salesApi.getPoclValidationErrorsForProcessing.mockResolvedValueOnce([poclValidationError])
+      await processPoclValidationErrors()
+
+      const [[[createTransactionPayload]]] = salesApi.createTransactions.mock.calls
+      expect(createTransactionPayload.dataSource).toBeUndefined()
+    })
+
+    it('leaves the datatype blank if payment source is not set', async () => {
```
Comment (irisfaraway) [inline packages/pocl-job/src/staging/__tests__/pocl-validation-errors.spec.js:257]:
```suggestion
    it('leaves the dataSource blank if payment source is not set', async () => {
```

**Review by irisfaraway — COMMENTED** [review]
👏 

Just a few nitpicks about naming tests, and also renaming the `UV` in variables to `Unvalidated` for clarity as discussed

**Comment by irisfaraway** [comment]
@jaucourt I can't approve since I started the PR, but LGTM! Happy for you to merge once the CI passes :shipit:

**Pull request merged**